### ISSUE TITLE
`programs.ssh.forwardX11` vs `services.openssh.forwardX11`

### ISSUE THREAD

**Issue by danbst** [NixOS/nixpkgs#21332]
## Issue description

I've added `programs.ssh.forwardX11` to my config, but rebuild showed a warning:
```
Failed assertions:
- cannot enable X11 forwarding without setting XAuth location
error: unable to build all machine configurations
```
The trick is, `programs.ssh.setXAuthLocation` is already `true`.

At first I thought that module system is broken somewhere, but then I stumbled into `services.openssh.forwardX11` (system builds with it). Is `programs.ssh.forwardX11` deprecated?

### Steps to reproduce

Add `programs.ssh.forwardX11 = true;` to config, observe system doesn't build.

## Technical details

* System: NixOS 16.09

**Comment by mdaiter** [comment]
You're supposed to enable `services.openssh.fowardX11`, which in turn should automatically enable `programs.ssh.setXAuthLocation`.

https://github.com/NixOS/nixpkgs/commit/d69dce080daf2c7004a79950e73dba6545b9097e
"The intent is that programs.ssh.setXAuthLocation should be
automatically enabled if needed or if xauth is already available."

**Comment by danbst** [comment]
Ah, thanks, so it isn't fixed in 16.09, but fixed in master.
